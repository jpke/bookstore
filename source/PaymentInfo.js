import React from 'react';
import { StyleSheet,
         Text,
         TextInput,
         TouchableHighlight,
         View
       } from 'react-native';

export default class PaymentInfo extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      nameOnCard: this.props.creditCardInfo.nameOnCard,
      cardNumber: this.props.creditCardInfo.cardNumber,
      securityCode: this.props.creditCardInfo.securityCode
    };
  }

  navigate(routeName) {
    if(routeName === "HomeScreen") {
      this.props.navigator.popToTop(0);
    } else {
      this.props.navigator.push({
        name: routeName
      });
    }
  }

  updateState(text, item) {
    item === 'nameOnCard' && this.setState({nameOnCard: text});
    item === 'cardNumber' && this.setState({cardNumber: text});
    item === 'securityCode' && this.setState({securityCode: text});
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.selectedBook}>
          <View style={styles.header}>
            <View style={{flex: 1}}>
              <TouchableHighlight style={styles.homeButton} onPress={() => this.navigate('HomeScreen')}>
                <Text style={styles.menuItem}>Home</Text>
              </TouchableHighlight>
            </View>
            <Text style={styles.title}>Credit Card Info</Text>
            <View style={{flex: 1}}></View>
          </View>
            <View style={styles.inputForm}>
              <TextInput style={styles.input}
                value={this.state.nameOnCard}
                placeholder="Name on Card"
                onChangeText={(text) => this.updateState(text, 'nameOnCard')}/>
              <TextInput style={styles.input}
                value={this.state.cardNumber}
                placeholder="Card Number"
                onChangeText={(text) => this.updateState(text, 'cardNumber')}/>
              <TextInput style={styles.input}
                value={this.state.securityCode}
                placeholder="Security Code"
                onChangeText={(text) => this.updateState(text, 'securityCode')}/>
            </View>
        </View>
        <View style={styles.menu}>
          <TouchableHighlight style={styles.button} onPress={() => this.props.navigator.pop()}>
            <Text style={styles.menuItem}>Back</Text>
          </TouchableHighlight>
          <TouchableHighlight style={styles.button} onPress={() => {
              this.props.submitInfo({
                nameOnCard: this.state.nameOnCard,
                cardNumber: this.state.cardNumber,
                securityCode: this.state.securityCode
              }, 'PaymentInfo');
              this.navigate('ReviewOrder');}
          }>
            <Text style={styles.menuItem}>Review Order</Text>
          </TouchableHighlight>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  header: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'flex-start',
    marginTop: 10,
  },
  selectedBook: {
    flex: 6,
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  title: {
    flex: 4,
    fontSize: 40,
    color: '#ffcd67',
    textShadowColor: 'black',
    textShadowOffset: {width: 1, height: 1},
    textShadowRadius: 4,
    textAlign: 'center'
  },
  inputForm: {
    flex: 3,
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  input: {
    flex: 1,
    maxHeight: 70,
    width: 250,
    fontSize: 18,
    borderWidth: 1,
    borderColor: '#ffcd67',
    backgroundColor: '#ffcd67',
    margin: 1,
    padding: 5,
    borderRadius: 5
  },
  bookAuthor: {
    fontSize: 15,
    padding: 5
  },
  menu: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  menuItem: {
    fontSize: 18,
    color: '#FFFFFF',
  },
  homeButton: {
    backgroundColor: '#6799FF',
    width: 80,
    margin: 5,
    paddingTop: 10,
    paddingBottom: 10,
    borderRadius: 10,
    alignItems: 'center',
  },
  button: {
    backgroundColor: '#6799FF',
    width: 150,
    padding: 10,
    margin: 10,
    borderRadius: 10,
    alignItems: 'center',
  }
});
