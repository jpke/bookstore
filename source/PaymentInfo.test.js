import React from 'react';
import ReactNative from 'react-native';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from './store';
const store = configureStore();

import PaymentInfo from './PaymentInfo';

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <PaymentInfo />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
