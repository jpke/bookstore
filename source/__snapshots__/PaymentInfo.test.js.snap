// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 6,
        "justifyContent": "space-around",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "#6799FF",
                "borderRadius": 10,
                "margin": 5,
                "paddingBottom": 10,
                "paddingTop": 10,
                "width": 80,
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
              }
            }
          >
            Home
          </Text>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#ffcd67",
            "flex": 4,
            "fontSize": 40,
            "textAlign": "center",
            "textShadowColor": "black",
            "textShadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "textShadowRadius": 4,
          }
        }
      >
        Credit Card Info
      </Text>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
          "justifyContent": "space-around",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Name on Card"
        style={
          Object {
            "backgroundColor": "#ffcd67",
            "borderColor": "#ffcd67",
            "borderRadius": 5,
            "borderWidth": 1,
            "flex": 1,
            "fontSize": 18,
            "margin": 1,
            "maxHeight": 70,
            "padding": 5,
            "width": 250,
          }
        }
        value=""
      />
      <TextInput
        onChangeText={[Function]}
        placeholder="Card Number"
        style={
          Object {
            "backgroundColor": "#ffcd67",
            "borderColor": "#ffcd67",
            "borderRadius": 5,
            "borderWidth": 1,
            "flex": 1,
            "fontSize": 18,
            "margin": 1,
            "maxHeight": 70,
            "padding": 5,
            "width": 250,
          }
        }
        value=""
      />
      <TextInput
        onChangeText={[Function]}
        placeholder="Security Code"
        style={
          Object {
            "backgroundColor": "#ffcd67",
            "borderColor": "#ffcd67",
            "borderRadius": 5,
            "borderWidth": 1,
            "flex": 1,
            "fontSize": 18,
            "margin": 1,
            "maxHeight": 70,
            "padding": 5,
            "width": 250,
          }
        }
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#6799FF",
            "borderRadius": 10,
            "margin": 10,
            "padding": 10,
            "width": 150,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
          }
        }
      >
        Back
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#6799FF",
            "borderRadius": 10,
            "margin": 10,
            "padding": 10,
            "width": 150,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
          }
        }
      >
        Review Order
      </Text>
    </View>
  </View>
</View>
`;
